tracer/test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs

```diff
@@ -432,18 +457,62 @@ public string GetTargetFramework()
             return $"net{_major}{_minor}{_patch ?? string.Empty}";
         }
 
-        public MockTracerAgent GetMockAgent(bool useStatsD = false)
+        public void EnableWindowsNamedPipes(string tracePipeName = null, string statsPipeName = null)
         {
-            // Strategy pattern for agent transports goes here
-            var agentPort = TcpPortProvider.GetOpenPort();
-            var agent = new MockTracerAgent(agentPort, useStatsd: useStatsD);
+            TransportType = TestTransports.WindowsNamedPipe;
+        }
+
+        public void EnableDefaultTcp()
+        {
+            TransportType = TestTransports.Tcp;
+        }
+
+        public void EnableUnixDomainSockets()
+        {
+#if NETCOREAPP
+            TransportType = TestTransports.Uds;
+#else
+            // Unsupported
+            TransportType = TestTransports.Tcp;
+#endif
+        }
 
-            _output.WriteLine($"Assigned port {agent.Port} for the agentPort.");
+        public MockTracerAgent GetMockAgent(bool useStatsD = false)
```

Comment:
Maybe you could pass the `TransportType` as an argument here, so the MockTracerAgent saves the `TransportType` field instead of the EnvironmentHelper?

Reply:
Good call

Reply:
Well, on second thought (and this came up in the PR where this method was introduced) ... I really like the idea of having the transport be transparent here and determined by the EnvironmentHelper as that is part of the purpose of the EnvironmentHelper: to facilitate setup/management of the "environment" a test case is running within.

This allows us to do things like run the whole test suite for any transport now or in the future if we want to do that as a one-off or on any sort of schedule.

Reply:
That's a good idea, let's keep this as-is in the EnvironmentHelper 👍🏼